Automattic/wp-calypso
# Show theme type on details page

Fixes https://github.com/Automattic/wp-calypso/issues/83149

## Proposed Changes

Some themes have a badge on the theme showcase, but nothing is shown on the theme details page. This PR changes that to show the Partner, Community, WooCommerce, and Sensei badges besides the theme title on the theme details page:

|Before|After|
|---|---|
|![image](https://github.com/Automattic/wp-calypso/assets/8511199/adb5d60f-4410-47e9-9e25-ae79bc3e740c)|![image](https://github.com/Automattic/wp-calypso/assets/8511199/4abea431-ada4-47b8-a3c5-1463c904d663)|

Design: RcxiuWYjZGkFRdlWL7L02m-fi-2092_37729

## Testing Instructions

Check that the themes details show a badge with the theme type:

- Community theme: `/theme/astra/{ siteSlug }.wordpress.com`
- Partner theme: `/theme/yuna/{ siteSlug }.wordpress.com`
- Sensei theme: `/theme/course/{ siteSlug }.wordpress.com`
- Woocommerce theme: `/theme/amulet/{ siteSlug }.wordpress.com`

Both as is, and with the `themes/tiers` flag (`?flags=themes/tiers`).

Navigate across the theme showcase and check that themes not belonging to any of the previous four categories show the badge next to the title.

## Pre-merge Checklist
